src/GafferImage/Catalogue.cpp

```diff
@@ -1131,14 +1148,37 @@ void Catalogue::imageReceived( Gaffer::Plug *plug )
 	internalImage->driverClosed();
 }
 
+
+void Catalogue::plugSet( const Plug *plug )
+{
+	// Enforce that only one image may have a particular output index
```

Comment:
It'd be nice to document the rationale for having `Image.outputIndex` rather than `Catalogue.outputIndices` - it being easier to enforce uniqueness than it is to sync indices during deletion/reordering. Could also add a todo to handle selection the same way?